cmd/zed/zed_conf.h

```diff
@@ -25,6 +25,7 @@ struct zed_conf {
 	unsigned	do_memlock:1;		/* true if locking memory */
 	unsigned	do_verbose:1;		/* true if verbosity enabled */
 	unsigned	do_zero:1;		/* true if zeroing state */
+	unsigned	do_testmode:1;		/* true if testmode is set */
```

Comment:
Rather than a `-t` test mode, what do you think about explicitly providing the PATH as via a command line option, `-P <path>`.  What I like about this is that it's much less vague (although the behavior was nicely explained in the man page), and it should let you simplify the code in `_zed_event_add_env_restrict`.